src/common/liboptparse/optparse.c

```diff
@@ -983,9 +1000,12 @@ optparse_err_t optparse_add_option_table (optparse_t *p,
     optparse_err_t rc = OPTPARSE_SUCCESS;
     const struct optparse_option *o = opts;
 
+    if (!p)
+        return OPTPARSE_BAD_ARG;
+
     while (o->usage) {
         if ((rc = optparse_add_option (p, o++)) != OPTPARSE_SUCCESS) {
-            option_table_remove (p, opts, o);
+            option_table_remove (p, opts, --o);
```

Comment:
i see the bug you caught here.  But should it be in a different commit?

Reply:
Good catch. Yeah I forgot I fixed that along the way. I'll split this out.

Reply:
Ok, latest push should have that fix split out into its own commit.

Reply:
LGTM, but did you mean "Fix the value for o (letter o) passed to ..."?  You did "0" (zero) instead.

Reply:
Thanks, fixed.